x/mint/test/mint_test.go

```diff
@@ -0,0 +1,118 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	minttypes "github.com/celestiaorg/celestia-app/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	accounts []string
+	cctx     testnode.Context
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	if testing.Short() {
+		s.T().Skip("skipping mint integration test in short mode.")
+	}
+
+	s.T().Log("setting up mint integration test suite")
+	s.accounts, s.cctx = testnode.DefaultNetwork(s.T())
+}
+
+// TestTotalSupplyIncreasesOverTime tests that the total supply of tokens
+// increases over time as new blocks are added to the chain.
+func (s *IntegrationTestSuite) TestTotalSupplyIncreasesOverTime() {
+	require := s.Require()
+
+	err := s.cctx.WaitForNextBlock()
+	require.NoError(err)
+	initalSupply := s.GetTotalSupply()
+
+	_, err = s.cctx.WaitForHeight(20)
+	require.NoError(err)
+	laterSupply := s.GetTotalSupply()
+
+	require.True(initalSupply.AmountOf(app.BondDenom).LT(laterSupply.AmountOf(app.BondDenom)))
+}
+
+// TestInitialInflationRate tests that the initial inflation rate is
+// approximately 8% per year.
+func (s *IntegrationTestSuite) TestInitialInflationRate() {
+	require := s.Require()
+
+	oneYear, err := time.ParseDuration(fmt.Sprintf("%vns", minttypes.NanosecondsPerYear))
+	require.NoError(err)
+
+	err = s.cctx.WaitForNextBlock()
+	require.NoError(err)
+	initialSupply := s.GetTotalSupply()
+	initialTimestamp := s.GetTimestamp()
+
+	_, err = s.cctx.WaitForHeight(20)
+	require.NoError(err)
+	laterSupply := s.GetTotalSupply()
+	laterTimestamp := s.GetTimestamp()
```

Comment:
this could end up using a different time stamp than when we got the supply, so if we want to minimize potential false negatives, then I think we actually have to download the latest header, then use the height from that header to query the state at a specific height.

for that, we might have to have a node that doesn't prune state, which we can configure in the app config

Reply:
I have yet to see any tho, so maybe we just write some comments to document it if it ever actually becomes an issue

Reply:
Resolved via 5b4fdca96a533b08cbc2ce45ce52d8b19a5fafff and 5dd630f1d9d93353544d9c0412b0171c65f5cb22